test_provider.py

```diff
@@ -0,0 +1,310 @@
+import json
+from datetime import datetime, timedelta
+from common.api import ProgramProvider
+from common.minimodel import *
+
+
+class JsonProvider(ProgramProvider):
+
+    # TODO: To handle Enum it could be done this way or changing the Enum to accept the 
+    # string value and just use the constructor. Let me know which is better.
+
+    obs_classes = {'partnerCal': ObservationClass.PARTNER_CAL,
```

Comment:
For `ObservationClasses`, we do want the implicit ordering that `auto` gives us, so there are two ways I can see this going. The first way is this way, which is fine. The second way requires a bit more thought, so let me get back to you on that Sunday or Monday.